pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -1084,7 +1092,8 @@ func KeyV2(
 		}
 	}
 
-	if nodeLocalDNSEnabled {
+	if version.ConstraintK8sLess134.Check(kubernetesVersion) && nodeLocalDNSEnabled ||
+		kubeProxyConfig != nil && kubeProxyConfig.Enabled != nil && *kubeProxyConfig.Enabled && kubeProxyConfig.Mode != nil && *kubeProxyConfig.Mode == gardencorev1beta1.ProxyModeIPVS && nodeLocalDNSEnabled {
```

Comment:
We should probably again use the extracted helper function to make this more readable.